### DISCUSSION TITLE
ROHD on DartPad, Codespaces, and Dev Containers

### DISCUSSION THREAD

**Discussion by mkorbel1** [intel/rohd#272] [Announcements]
We're excited to share that there are some new ways to easily develop with ROHD!

### DartPad
ROHD is now included as one of the available packages in DartPad (https://dartpad.dev/)!  Think of it a bit like EDA Playground, but for Dart.  You can use GitHub gists to share code snippets on DartPad (https://github.com/dart-lang/dart-pad/wiki/Sharing-Guide).

### GitHub Codespaces & devcontainer.json
You can open the ROHD, ROHD-VF, and ROHD Cosim repositories in a pre-configured GitHub Codespace (https://github.com/features/codespaces) or development container (https://code.visualstudio.com/docs/devcontainers/containers) with everything pre-installed, cloned, and configured.  Even free GitHub accounts get some amount free on GitHub Codespaces per month, so it's really easy to try things out right within your browser!  There are convenient buttons added to the top of all three repositories to open a new GitHub Codespace for the repo.

These are a great way to experiment with ROHD or make contributions without needing to install anything on your local system.

**Comment by chykon** [comment]
I did not know that ROHD can now run in DartPad. Cool! A handy thing for interactive examples and does not require an account, unlike GitHub Codespaces. I ran the examples - they work, except for any operations with the file system, of course.

Do `development containers` work well outside of Codespaces? Does everything run without problems?

**Reply by mkorbel1** [reply]
That's right, DartPad has some limitations.

Yes, the dev containers work outside of Codespaces, though keep in mind you'll need to have Docker installed.  Also, if you work behind a VPN, you may have to ensure your proxies are set up properly to download requirements to build the container.
